HyperVExtension/src/HyperVExtension/Models/VMGalleryCreationAdaptiveCardSession.cs

```diff
@@ -0,0 +1,341 @@
+            {
+                return new ProviderOperationResult(ProviderOperationStatus.Failure, null, "Failed to get review form", "Selected image index is out of range");
+            }
+
+            var galleryImage = _vMGalleryImageList.Images[inputForGalleryOperation.SelectedImageListIndex];
+            var newVirtualMachineNameLabel = _stringResource.GetLocalized("NameLabelForNewVirtualMachine", ":");
+            var primaryButtonForCreationFlowText = _stringResource.GetLocalized("PrimaryButtonLabelForCreationFlow");
+            var secondaryButtonForCreationFlowText = _stringResource.GetLocalized("SecondaryButtonLabelForCreationFlow");
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(Constants.ExtensionIconInternal));
+            var randomAccessStream = await storageFile.OpenReadAsync();
+
+            // Convert the stream to a byte array
+            var bytes = new byte[randomAccessStream.Size];
+            await randomAccessStream.ReadAsync(bytes.AsBuffer(), (uint)randomAccessStream.Size, InputStreamOptions.None);
+            var providerBase64Image = Convert.ToBase64String(bytes);
+            var reviewFormData = new JsonObject
+            {
+                { "ProviderName", HyperVStrings.HyperVProviderDisplayName },
+                { "DiskImageSize", BytesHelper.ConvertBytesToString(galleryImage.Disk.SizeInBytes) },
+                { "VMGalleryImageName", galleryImage.Name },
+                { "Publisher", galleryImage.Publisher },
+                { "NameOfNewVM", inputForGalleryOperation.NewVirtualMachineName },
+                { "NameLabel", newVirtualMachineNameLabel },
+                { "Base64ImageForProvider", providerBase64Image },
+                { "DiskImageUrl", galleryImage.Symbol.Uri },
+                { "PrimaryButtonLabelForCreationFlow", primaryButtonForCreationFlowText },
+                { "SecondaryButtonLabelForCreationFlow", secondaryButtonForCreationFlowText },
+            };
+
+            var template = LoadTemplate(SessionState.ReviewForm);
+
+            return _creationAdaptiveCard!.Update(LoadTemplate(SessionState.ReviewForm), reviewFormData.ToJsonString(), "reviewForm");
+        }
+        catch (Exception ex)
+        {
+            var reviewFormGenerationError = _stringResource.GetLocalized("ReviewFormGenerationFailedError");
+            return new ProviderOperationResult(ProviderOperationStatus.Failure, ex, reviewFormGenerationError, ex.Message);
+        }
+    }
+
+    /// <summary>
+    /// The discription for VM gallery images is stored in a list of strings. This method merges the strings into one string.
+    /// </summary>
+    /// <param name="image">The c# class that represents the gallery image</param>
+    /// <returns>A string that combines the original list of strings into one</returns>
+    public string GetMergedDescription(VMGalleryImage image)
+    {
+        var description = string.Empty;
```

Comment:
If I have list of strings like:
    "This is a VM description."
    "This is\r\nanother VM\r\ndescription."
that will become
    "This is a VM description.This isanother VMdescription."

No separation between strings and enw lines at all. Is that intended? string.Empty is ... empty, so no separators.

Reply:
I was going off of what is currently in the VM gallery Json today. Interestingly It actually appears like this

```Json
"description": [
  "Ubuntu is used by thousands of development teams around the world because of its versatility, reliability and extensive developer libraries.",
  "\r\n\r\n",
  "This LTS release comes with five years of support meaning that you can be assured of secure and dependable operating system giving you the fastest route from development to deployment on desktop, devices, server, cloud and IoT edge.",
  "\r\n\r\n",
  "Ubuntu is free and will always be, and you have the option to get support and Landscape from Canonical. See https://www.ubuntu.com/legal for terms and conditions."
],
```

its a list of strings and between each string of text they hard code this line `"\r\n\r\n"` as an item in the list. So thats why I ended up using string.Empty. I think we're ok here since these haven't changes in years. I'll see what Hart intends to add when he gets there and if I'll need to update this or not.

Reply:
Then having a space between non-empty array items would be still beneficial.
Could you add that string array example as a comment, so we know what it looks like?
Also, why not just return "description" on the last line? string.Join(string.Empty, description) doesn't do anything other than copy "description".